com.unity.multiplayer.mlapi/Editor/ProfilerModuleOverride/NetworkProfilerOverride.cs

```diff
@@ -0,0 +1,249 @@
+            list.Add(new MLAPIProfilerCounter() { m_Name = ProfilerConstants.NumberOfRPCsInQueueSize.ToString(), m_Category = ProfilerCategory.Network.Name });
+            list.Add(new MLAPIProfilerCounter() { m_Name = ProfilerConstants.NumberOfRPCsOutQueueSize.ToString(), m_Category = ProfilerCategory.Network.Name });
+            return list;
+        }
+
+        private static List<MLAPIProfilerCounter> CreateOperationsCounters()
+        {
+            var list = new List<MLAPIProfilerCounter>();
+            list.Add(new MLAPIProfilerCounter() { m_Name = ProfilerConstants.NumberOfConnections.ToString(), m_Category = ProfilerCategory.Network.Name });
+            list.Add(new MLAPIProfilerCounter() { m_Name = ProfilerConstants.ReceiveTickRate.ToString(), m_Category = ProfilerCategory.Network.Name });
+            return list;
+        }
+
+        private static List<MLAPIProfilerCounter> CreateMessagesCounters()
+        {
+            var list = new List<MLAPIProfilerCounter>();
+            list.Add(new MLAPIProfilerCounter() { m_Name = ProfilerConstants.NumberOfNamedMessages.ToString(), m_Category = ProfilerCategory.Network.Name });
+            list.Add(new MLAPIProfilerCounter() { m_Name = ProfilerConstants.NumberOfUnnamedMessages.ToString(), m_Category = ProfilerCategory.Network.Name });
+            list.Add(new MLAPIProfilerCounter() { m_Name = ProfilerConstants.NumberBytesSent.ToString(), m_Category = ProfilerCategory.Network.Name });
+            list.Add(new MLAPIProfilerCounter() { m_Name = ProfilerConstants.NumberBytesReceived.ToString(), m_Category = ProfilerCategory.Network.Name });
+            list.Add(new MLAPIProfilerCounter() { m_Name = ProfilerConstants.NumberNetworkVarsReceived.ToString(), m_Category = ProfilerCategory.Network.Name });
+            return list;
+        }
+
+        private delegate List<MLAPIProfilerCounter> CounterFunction();
+
+        private static bool UpdateOrCreateModule(ref MLAPIModules mlapiModules, string moduleName, CounterFunction counterFunction)
+        {
+            bool hasChanged = false;
+            var module = mlapiModules.m_Modules.Find(x => x.m_Name == moduleName);
+            if (module == null)
+            {
+                var newModule = new MLAPIProfilerModuleData();
+                newModule.m_Name = moduleName;
+                newModule.m_ChartCounters = counterFunction();
+                newModule.m_DetailCounters = counterFunction();
+                mlapiModules.m_Modules.Add(newModule);
+                hasChanged = true;
+            }
+            else
+            {
+                var counters = counterFunction();
+                if (module.m_ChartCounters.Count != counters.Count || module.m_DetailCounters.Count != counters.Count)
+                {
+                    module.m_ChartCounters = counters;
+                    module.m_DetailCounters = counters;
+                    hasChanged = true;
+                }
+            }
```

Comment:
A little confused by this. In one branch you're calling `counterFunction` twice, and the other you're calling it once. this means that sometimes it's sharing the counters list and other times they're distinct. sounds like the makings for weird bugs...

Reply:
This part has been changed